### PR TITLE
Chat monitor: Improve the handling of usernames

### PR DESCRIPTION
**Pull request by AnnikaCodes**
This PR ignores word boundaries in evasion regexes when filtering usernames, preventing the use of usernames with evasions of slurs in them.

I also added a unit test for this behavior, to avoid regressions.

### CODE REVIEW

**Comment by AnnikaCodes** [comment]
@Zarel  bump

**Review by Zarel — APPROVED** [review]
Looks good! You can merge this without further review once you handle the merge conflict.

**Pull request merged**
